app/validators/aus_phone_number_validator.rb

```diff
@@ -0,0 +1,7 @@
+class AusPhoneNumberValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    if value != nil && value !~ /\A04[0-9]{8}\z/
```

Comment:
This is a little pedantic. It would be much nicer to allow the user to put spaces in their number and just strip them server side.